shared/js/background/components/email-autofill.js

```diff
@@ -0,0 +1,353 @@
+        return { alias: userData?.nextAlias }
+    }
+
+    /**
+     * @returns {Promise<import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').RefreshAliasResponse>}
+     */
+    async refreshAlias () {
+        await this.fetchAlias()
+        return this.getAddresses()
+    }
+
+    getAddresses () {
+        const userData = this.settings.getSetting('userData')
+        return {
+            personalAddress: userData?.userName,
+            privateAddress: userData?.nextAlias
+        }
+    }
+
+    showContextMenuAction () {
+        if (this.contextMenuAvailable) {
+            browser.contextMenus.update(MENU_ITEM_ID, { visible: true })
+        }
+    }
+
+    hideContextMenuAction () {
+        if (this.contextMenuAvailable) {
+            browser.contextMenus.update(MENU_ITEM_ID, { visible: false })
+        }
+    }
+
+    /**
+     *
+     * @param {FirePixelOptions}  options
+     */
+    sendJSPixel (options) {
+        const { pixelName } = options
+        switch (pixelName) {
+        case 'autofill_show':
+            this.fireAutofillPixel('email_tooltip_show_extension')
+            break
+        case 'autofill_private_address':
+            this.fireAutofillPixel('email_filled_random_extension', true)
+            break
+        case 'autofill_personal_address':
+            this.fireAutofillPixel('email_filled_main_extension', true)
+            break
+        case 'incontext_show':
+            fireIncontextSignupPixel('incontext_show_extension')
+            break
+        case 'incontext_primary_cta':
+            fireIncontextSignupPixel('incontext_primary_cta_extension')
+            break
+        case 'incontext_dismiss_persisted':
+            fireIncontextSignupPixel('incontext_dismiss_persisted_extension')
+            break
+        case 'incontext_close_x':
+            fireIncontextSignupPixel('incontext_close_x_extension')
+            break
+        default:
+            getFromSessionStorage('dev').then(isDev => {
+                if (isDev) console.error('Unknown pixel name', pixelName)
+            })
+        }
+    }
+
+    fireAutofillPixel (pixel, shouldUpdateLastUsed = false) {
+        const browserName = getBrowserName() ?? 'unknown'
+        if (!pixelsEnabled) return
+
+        const userData = this.settings.getSetting('userData')
+        if (!userData?.userName) return
+
+        const lastAddressUsedAt = this.settings.getSetting('lastAddressUsedAt') ?? ''
+
+        sendPixelRequest(getFullPixelName(pixel, browserName), { duck_address_last_used: lastAddressUsedAt, cohort: userData.cohort })
+        if (shouldUpdateLastUsed) {
+            this.settings.updateSetting('lastAddressUsedAt', currentDate())
+        }
+    }
+
+    getIncontextSignupDismissedAt () {
+        const permanentlyDismissedAt = this.settings.getSetting('incontextSignupPermanentlyDismissedAt')
+        // TODO: inject this dependency (after TDS refactor lands)
```

Comment:
There are a couple of these. I'm not quite sure who's supposed to follow-up. Is it an action item to whomever is doing that TDS refactor?

Reply:
That's for me, referring to https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/2309